probe/endpoint/resolver_internal_test.go

```diff
@@ -0,0 +1,41 @@
+package endpoint
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestReverseResolver(t *testing.T) {
+	tests := map[string]string{
+		"8.8.8.8": "google-public-dns-a.google.com",
+		"8.8.4.4": "google-public-dns-b.google.com",
+	}
```

Comment:
We need to mock out the actual dns lookup - do you mind if I have a stab at that?